tests/syntax/test_pruning.py

```diff
@@ -21,6 +22,57 @@ def test_containment_in():
     assert any(0.5 <= x <= 0.7 or 1.3 <= x <= 1.5 for x in xs)
 
 
+def test_containment_2d_region():
+    """Test pruning based on object containment in a 2D region.
+
+    Specifically tests that vertical portions of baseOffset are not added
+    to maxDistance, and that if objects are known to be flat in the plane,
+    their height is not considered as part of the minRadius.
+    """
+    # Tests the effect of the vertical portion of baseOffset in a 2D region.
+    scenario = compileScenic(
+        """
+        workspace = Workspace(PolygonalRegion([0@0, 2@0, 2@2, 0@2]))
+        ego = new Object on workspace
+    """
+    )
+    # Sampling should only require 1 iteration after pruning
+    xs = [sampleEgo(scenario).position.x for i in range(60)]
+    assert all(0.5 <= x <= 1.5 for x in xs)
+    assert any(0.5 <= x <= 0.7 or 1.3 <= x <= 1.5 for x in xs)
+
+    # Test height's effect in a 2D region.
+    scenario = compileScenic(
+        """
+        workspace = Workspace(PolygonalRegion([0@0, 2@0, 2@2, 0@2]))
+        ego = new Object in workspace, with height 0.1
+    """
+    )
+    # Sampling should only require 1 iteration after pruning
+    xs = [sampleEgo(scenario).position.x for i in range(60)]
+    assert all(0.5 <= x <= 1.5 for x in xs)
+    assert any(0.5 <= x <= 0.7 or 1.3 <= x <= 1.5 for x in xs)
+
+    # Test both combined, in a slightly more complicated case.
+    # Specifically, there is a non vertical component to baseOffset
+    # that should be accounted for and the height is random.
+    scenario = compileScenic(
+        """
+        class TestObject:
+            baseOffset: (0.1, 0, self.height/2)
+
+        workspace = Workspace(PolygonalRegion([0@0, 2@0, 2@2, 0@2]))
+        ego = new Object in workspace, with height Range(0.1,0.5)
+    """
+    )
+    # Sampling should fail ~30.56% of the time, so
```

Comment:
Could you explain why this is the case? Since the non-vertical component of the offset is constant I would have thought pruning could take it into account exactly.